src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,645 @@
+                    .chain(Some(("z_84_alpha_check", z_84_alpha_check)))
+                    .chain(Some(("alpha_0_check", alpha_0.clone() - expected_alpha_0)))
+            };
+
+            // Check α_0_prime = α_0 + 2^130 - t_p
+            let alpha_0_prime_check = {
+                let two_pow_130 = Expression::Constant(pallas::Base::from_u128(65).square());
+                let t_p = Expression::Constant(pallas::Base::from_u128(T_P));
+                alpha_0_prime - (alpha_0 + two_pow_130 - t_p)
+            };
+
+            // We want to enforce canonicity of a 255-bit base field element, α.
+            // That is, we want to check that 0 ≤ α < p, where p is Pallas base
+            // field modulus p = 2^254 + t_p
+            //                 = 2^254 + 45560315531419706090280762371685220353.
+            // Note that t_p < 2^130.
+            //
+            // α has been decomposed into three pieces,
+            //            α = α_0 (252 bits)  || α_1 (2 bits) || α_2 (1 bit).
+            //
+            // If the MSB α_2 = 1, then:
+            //      - α_2 = 1 => α_1 = 0, and
+            //      - α_2 = 1 => α_0 < t_p. To enforce this:
+            //          - α_2 = 1 => 0 ≤ α_0 < 2^130
+            //                    => 13 ten-bit lookups of α_0
+            //          - α_2 = 1 => 0 ≤ α_0 + 2^130 - t_p < 2^130
+            //                    => 13 ten-bit lookups of α_0 + 2^130 - t_p
+            //                    => z_13_alpha_0_prime = 0
+            //
+            let canon_checks = {
+                std::iter::empty()
+                    .chain(Some(("MSB = 1 => alpha_1 = 0", alpha_2.clone() * alpha_1)))
+                    .chain(Some((
+                        "MSB = 1 => z_13_alpha_0 = 0",
+                        alpha_2.clone() * z_13_alpha_0,
+                    )))
+                    .chain(Some((
+                        "MSB = 1 => z_13_alpha_0_prime = 0",
+                        alpha_2 * z_13_alpha_0_prime,
+                    )))
+            };
+
+            canon_checks
+                .chain(decomposition_checks)
+                .chain(Some(("z_85_alpha = 0", z_85_alpha)))
+                .chain(Some(("alpha_0_prime check", alpha_0_prime_check)))
+                .map(move |(name, poly)| (name, base_field_fixed_canon.clone() * poly))
+        });
+    }
+
+    pub fn assign(
+        &self,
+        mut layouter: impl Layouter<pallas::Base>,
+        scalar: CellValue<pallas::Base>,
+        base: OrchardFixedBasesFull,
+    ) -> Result<EccPoint, Error> {
+        let (result, scalar) = layouter.assign_region(
+            || "Base-field elem fixed-base mul",
+            |mut region| {
+                let offset = 0;
+
+                // Decompose scalar
+                let scalar = self.decompose_base_field_elem(scalar, offset, &mut region)?;
+
+                let (acc, mul_b) = self.super_config.assign_region_inner(
```

Comment:
How does this work? `decompose_base_field_elem` assigns a running sum into the `window` column, but `super_config.assign_region_inner` requires that this column contain the 3-bit words themselves, as it uses the cells directly to interpolate `x`.

Reply:
This works because the `mul_fixed::Config::coords_check()` creates the gate that checks interpolation, and this helper takes in an expression for `window`.

In `base_field_elem::Config::create_gate()`, we input the expression 
```
word = z_cur - z_next * pallas::Base::from_u64(constants::H as u64);
```
as the "window" to the `coords_check()` helper.